specs/test_formats/README.md

```diff
@@ -0,0 +1,170 @@
+
+## Glossary
+
+- `generator`: a program that outputs one or more `suite` files.
+  - A generator should only output one `type` of test.
+  - A generator is free to output multiple `suite` files, optionally with different `handler`s.
+- `type`: the specialization of one single `generator`.
+- `suite`: a YAML file with:
+  - a header: describes the `suite`, and defines what the `suite` is for
+  - a list of test cases
+- `runner`: where a generator is a *"producer"*, this is the *"consumer"**.
+  - A `runner` focuses on *only one* `type`, and each type has *only one* `runner`.
+- `handler`: a `runner` may be too limited sometimes, you may have a `suite` with a specific focus that requires a different format.
+  To facilitate this, you specify a `handler`: the runner can deal with the format by using the specified handler.
+  Using a `handler` in a `runner` is optional.
+- `case`: a test case, an entry in the `test_cases` list of a `suite`. A case can be anything in general, 
+  but its format should be well-defined in the documentation corresponding to the `type` (and `handler`).\
+  A test has the same exact configuration and fork context as the other entries in the `case` list of its `suite`.
+- `forks_timeline`: a fork timeline definition, a YAML file containing a key for each fork-name, and a slot number as value.
+
+## Test format philosophy
+
+### Config design
+
+After long discussion, the following types of configured constants were identified:
+- Never changing: genesis data
+- Changing, but reliant on old value: e.g. a slot time may change, but if you want to do the conversion 
+  `(genesis data, timestamp) -> slot number` you end up needing both constants.
+- Changing, but kept around during fork transition: finalization may take a while,
+  e.g. an executable has to deal with new deposits and old deposits at the same time. Another example may be economic constants.
+- Additional, back-wards compatible: new constants are introduced for later phases
+- Changing: there is a very small chance some constant may really be *replaced*. 
+  In this off-chance, it is likely better to include it as an additional variable,
+  and some clients may simply stop supporting the old one, if they do not want to sync from genesis.
+
+Based on these types of changes, we model the config as a list of key value pairs,
+ that only grows with every fork (they may change in development versions of forks however, git manages this).
+With this approach, configurations are backwards compatible (older clients ignore unknown variables), and easy to maintain.
+
+### Fork config design
+
+There are two types of fork-data:
+1) timeline: when does a fork take place?
+2) coverage: what forks are covered by a test?
+
+The first is neat to have as a separate form: we prevent duplication, and can run with different presets
+ (e.g. fork timeline for a minimal local test, for a public testnet, or for main-net)
```

Comment:
It sounds more like "Chain config" to me. What's the use case here?

Reply:
See issue comment on genesis format / timelines. Let me know if there's any way to improve it, without making the format too heavy, or too far from an eth-2 type genesis.